packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -454,7 +455,8 @@ class ScrollableState<T extends Scrollable> extends State<T> {
   /// If there are no in-progress scrolling physics, this function scrolls to
   /// the given offset instead.
   void didUpdateScrollBehavior(double newScrollOffset) {
-    setState(() { /* The scroll behavior is part of our build state. */ });
+    // This does not call setState, because if anything below actually
```

Comment:
The only way that can happen (the contentExtents changing) is during layout, right? And if you're doing layout, you're guaranteed to repaint.